Add SLURM support for multi-node tests

roclark opened the pull request:
To make it easier to run on large clusters, Bobber should be able to run on SLURM clusters with Pyxis and Enroot installed. This would replace the need for mpirun and SSH keys/daemons inside the containers, making it easier to run tests without copying images between nodes or synchronizing SSH keys.

Closes #1

Signed-Off-By: Robert Clark <roclark@nvidia.com>

roclark committed:
Add SLURM support for multi-node tests

To make it easier to run on large clusters, Bobber should be able to run
on SLURM clusters with Pyxis and Enroot installed. This would replace the
need for mpirun and SSH keys/daemons inside the containers, making it
easier to run tests without copying images between nodes or synchronizing
SSH keys.

Signed-Off-By: Robert Clark <roclark@nvidia.com>

roclark commented:
This is currently a draft based on the ongoing discussion in #1. At this point, the NCCL tests should be fully functional using the Python wheel. As I see it, the following items are still required:

- [x] Add DALI tests
- [ ] Add FIO tests
- [x] Add mdtest
- [ ] Document the installation and usage
- [ ] Update the troubleshooting guide with steps to fix common issues
- [ ] Get a public image up on NGC

fredvx commented on bobber/lib/system/slurm.py:
```diff
@@ -0,0 +1,175 @@
+    installed on a system, the absolute path to this directory may change, but
+    the relative path is easy to find compared to this module. By allowing
+    Python to determine the absolute path to this module, the absolute path to
+    slurm_scripts can be found by combining the absollute path of this module
```
There's a typo in the word "absolute".

roclark committed:
Verify if Docker is running

While using Slurm, it is entirely possible to still use Bobber but not
have Docker installed on the head node where the jobs will be launched.
In this case, Docker should be ignored unless one of the commands
directly needs the Docker runtime.

Signed-Off-By: Robert Clark <roclark@nvidia.com>
